src/core/hle/service/apt/apt.cpp

```
@@ -203,78 +209,93 @@ void SendParameter(Service::Interface* self) {
 }
 
 void ReceiveParameter(Service::Interface* self) {
-    u32* cmd_buff = Kernel::GetCommandBuffer();
-    u32 app_id = cmd_buff[1];
-    u32 buffer_size = cmd_buff[2];
-    VAddr buffer = cmd_buff[0x104 >> 2];
-
-    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
-    cmd_buff[2] = next_parameter.sender_id;
-    cmd_buff[3] = next_parameter.signal;        // Signal type
-    cmd_buff[4] = next_parameter.buffer.size(); // Parameter buffer size
-    cmd_buff[5] = 0x10;
-    cmd_buff[6] = 0;
-    if (next_parameter.object != nullptr)
-        cmd_buff[6] = Kernel::g_handle_table.Create(next_parameter.object).MoveFrom();
-    cmd_buff[7] = (next_parameter.buffer.size() << 14) | 2;
-    cmd_buff[8] = buffer;
+    IPC::RequestParser rp(Kernel::GetCommandBuffer(), 0xD, 2, 0); // 0xD0080
+    u32 app_id = rp.Pop<u32>();
+    u32 buffer_size = rp.Pop<u32>();
+
+    size_t static_buff_size;
+    VAddr buffer = rp.PeekStaticBuffer(0, &static_buff_size);
```

>are there generic errors when the size of the input is too big for the allocated static buffer ? Is it the only error a service needs to check when sending data through static buffers ?

This kind of error is not checked by the service (because actual service doesn't read client command_buffer+0x100 at all). The error is checked by the kernel, and yes the error code is generic (I don't know the value though). There are other kind of errors like invalid address, which are also checked by the kernel and produce generic errors. Service doesn't check any thing here, so it is pretty safe to say any possible error code is generic